sbi/analysis/plot.py

```diff
@@ -1403,3 +1408,166 @@ def _plot_hist_region_expected_under_uniformity(
+        labels (list): list of labels for every density estimator.
+        colors (list): list of colors for every density estimator.
+        pp_vals_null (dict): dictionary of PP-values for each trial under the null
+            hypothesis.
+        ax (matplotlib.axes.Axes): axes to plot on.
+        **kwargs: keyword arguments for matplotlib.pyplot.plot.
+
+    Returns:
+        ax (matplotlib.axes.Axes): axes with the P-P plot.
+    """
+    if ax is None:
+        ax = plt.gca()
+    alphas = np.linspace(0, 1, 100)
+    pp_vals_dirac = PP_vals([0.5] * len(probas), alphas)
+    ax.plot(alphas, pp_vals_dirac, "--", color="black", label="True probability (H0)")
+
+    if pp_vals_null is None:
+        pp_vals_null = {}
+        for t in range(len(probas_null)):
+            pp_vals_null[t] = pd.Series(PP_vals(probas_null[t], alphas))
+
+    low_null = pd.DataFrame(pp_vals_null).quantile(conf_alpha / 2, axis=1)
+    up_null = pd.DataFrame(pp_vals_null).quantile(1 - conf_alpha / 2, axis=1)
+    ax.fill_between(
+        alphas,
+        low_null,
+        up_null,
+        color="grey",
+        alpha=0.2,
+        label=f"{(1 - conf_alpha) * 100}% confidence region",
+    )
+
+    for p, l_, c in zip(probas, labels, colors):
+        pp_vals = pd.Series(PP_vals(p, alphas))
+        ax.plot(alphas, pp_vals, label=l_, color=c, **kwargs)
+    return ax
+
+
+def pairplot_with_proba_intensity(
+    df_probas,
+    dim=1,
+    z_space=True,
+    n_bins=20,
+    vmin=0,
+    vmax=1,
+    cmap=cm.get_cmap("Spectral_r"),
+    show_colorbar=True,
+    ax=None,
+):
+    """Plot 1d or 2d marginal histogram of samples of the density estimator
+    with probabilities as color intensity.
+
+    Args:
+        df_probas (pd.DataFrame): dataframe with predicted class probabilities
+            as obtained from `compute_dfs_with_probas_marginals`.
+        dim (int): dimension of the marginal histogram to plot.
+        z_space (bool): whether to plot the histogram in latent space
+            of the base distribution of a normalizing flow. If False, plot
+            in the original space of the estimated density.
+        n_bins (int): number of bins for the histogram.
+        vmin (float): minimum value for the color intensity.
+        vmax (float): maximum value for the color intensity.
+        cmap (matplotlib.colors.Colormap): colormap for the color intensity.
+        show_colorbar (bool): whether to show the colorbar.
+        ax (matplotlib.axes.Axes): axes to plot on.
+
+    Returns:
+        ax (matplotlib.axes.Axes): axes with the plot.
+    """
+    if ax is None:
+        ax = plt.gca()
+
+    if dim == 1:
+        _, bins, patches = ax.hist(df_probas.z, n_bins, density=True, color="green")
+        df_probas["bins"] = np.select(
+            [df_probas.z <= i for i in bins[1:]], list(range(n_bins))
+        )
+        # get mean predicted proba for each bin
+        weights = df_probas.groupby(["bins"]).mean().probas
```

Comment:
you can also do this with https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binned_statistic.html

I have nothing agains pandas but I feel that numeric package should it as much as possible. Doing pure numpy / scipy operations will generally be faster.